java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/source/TaskToReceiverBridge.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.servicebus.source;
+
+import io.lenses.kcql.Kcql;
+import io.lenses.streamreactor.connect.azure.servicebus.config.AzureServiceBusConfigConstants;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+
+/**
+ * Bridge between Receivers and Connector's Task for Azure Service Bus.
+ */
+@Slf4j
+public class TaskToReceiverBridge {
+
+  private static final int INITIAL_RECORDS_TO_COMMIT_SIZE = 500;
+  private final BlockingQueue<ServiceBusMessageHolder> recordsQueue;
+  private final ServiceBusPartitionOffsetProvider offsetProvider;
+  private final ExecutorService sentMessagesExecutors;
+  private Map<String, ServiceBusReceiverFacade> receivers;
+  private final Map<String, ServiceBusMessageHolder> recordsToCommitMap;
+
+  /**
+   * Creates Bridge between Receivers and Connector's Task for Azure Service Bus.
+   * 
+   * @param properties          all properties for {@link org.apache.kafka.connect.source.SourceTask}.
+   * @param recordsQueue        records queue used to store received messages.
+   * @param offsetStorageReader offset storage reader from Task.
+   */
+  public TaskToReceiverBridge(Map<String, String> properties,
+      BlockingQueue<ServiceBusMessageHolder> recordsQueue, OffsetStorageReader offsetStorageReader) {
+    this.recordsQueue = recordsQueue;
+    this.offsetProvider = new ServiceBusPartitionOffsetProvider(offsetStorageReader);
+    String connectionString = properties.get(AzureServiceBusConfigConstants.CONNECTION_STRING);
```

Comment:
It is interesting idea, but I'm not sure I see the benefits clearly. The configuration will still be passed around, it's just gonna be form (POJO) that's different, isn't it?

Reply:
Single Responsibility Principle

We should have code that handles KCQL parsing and configuration concerns.  These should not be the same code as the connector logic.  Also it would reduce the duplication I identified around parsing the KCQL twice.

Reply:
It's about having single responsibility (being bridge between facade and task), not reducing the dependencies (constructor arguments). But I just moved logic around handling of connectionString and KCQL out of this class as you wished.